tests/async-unit/test_dual_mode_inference.py

```
@@ -0,0 +1,166 @@
+        exception retrieved by the done-callback (logged at DEBUG) -- pins
+        that the cancel-with-callback pattern exists, so asyncio never emits
+        "Task exception was never retrieved" at GC time."""
+        model = _dual_model()
+
+        async def fake_timeout_infer(*args, **kwargs):
+            if kwargs.get("model") == "fast-model":
+                return ("fast answer", [])
+            try:
+                await asyncio.sleep(STRAGGLER_SLEEP_S)
+            except asyncio.CancelledError:
+                raise RuntimeError("straggler exploded during cancellation")
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)